### PR TITLE
Script API: in FillSaveGameList() accept a range of slots

### PR DESCRIPTION
**Pull request by ivan-mogilko**
I don't like adding "last minute" things  to an upcoming release, but this request seemed easy enough to try.

Historically FillSaveGameList has a hardcoded range of 0 to 99 slots, and for some reason also additionally limits total shown slots to 50 items. This change lets user to tell which range of slots to fill, and removes item count limit, because it makes no sense anymore.

```
/// Fills the list box with the current user's saved games in the given range of slots.
import int  FillSaveGameList(int min_slot = 0, int max_slot = 99);
```

**Backwards compatibility**
Old compiled games will be using same old range (0-99) and 50 items limit.

### CODE REVIEW

**Comment by ericoporto** [comment]
This is related to #816 (I think)

**Comment by ivan-mogilko** [comment]
It is related, but the #816 was about solving this when running already existing games without rescripting them.

This pr only lets game authors to show more in their games. There may have to be a separate change to let override this limit in existing games.

**Comment by ericoporto** [comment]
> I might actually add the config option which hacks this number for testing / running old games, which will likely be enough to close

Not sure if we need another hack for changing old things, if it's working as is, it's fine, better to look into the future - less options are less surface for bugs.

**Comment by ivan-mogilko** [comment]
This is for running old games with the backwards compatible engine.

**Comment by ivan-mogilko** [comment]
I am actually going to suspend this, so converted to draft for the time being.

While the change to FillSaveGameList itself is really trivial, the consequence is that people may use any arbitrary range of saves.
And then this old DeleteSaveSlot behavior gets in a way:
https://adventuregamestudio.github.io/ags-manual/Globalfunctions_General.html#deletesaveslot
> for historical reasons, when deleting slots in the range of 1 to 50, this function also moves highest slot found in that range to fill the freed slot. For example, if there were slots 1, 2, 3, 4, 5 and DeleteSaveSlot removes slot 3, then save in slot 5 will be renamed to become slot 3.
See commit: https://github.com/adventuregamestudio/ags/pull/2289/commits/8e07fcd191ea55f9a9cee51cae4de07462346209

If I allow to list any arbitrary range of saves in the listbox, then this behavior must also be either adjusted or removed.
As a consequence, there's a chance this will break some game logic, because currently games might account for this, or even use this behavior on purpose. 
Which in turn may lead to users requesting additional setup or commands (like RenameSaveSlot, for a hypothetical example).
In other words, this may require some extra api design, which should not be done in a rush.

**Comment by ivan-mogilko** [comment]
I might remake this PR for ags4 instead, and later see if that is safe to backport or anything for some tiny 3.6 update.

**Comment by ivan-mogilko** [comment]
Implemented in ags4 by #2293. This may be later backported to ags3 branch if really wanted.
